Shopify/shopify-cli
# move dependency install before library version check

### WHY are these changes introduced?
While testing script pushes, I discovered that the `npm list` command that we use as part of getting the library version for `typescript` scripts fails if we haven't run `npm install` first.

Running `npm install` is typically always done during the `script create`. However, if you skip creation, and instead just get a script directly (i.e. the source is in a repo you clone), you run into this issue.

### WHAT is this pull request doing?
This PR moves the call to `ProjectDependencies.install` to before our call to `task_runner.library_version(library_name)`, ensuring that we don't attempt the `npm list` command without first installing the dependencies.

### How to test your changes?
To trigger the error:
- Take any typescript script and `rm -rf ./node_modules`
- `shopify-dev script create`
Then repeat with the this branch to see it work.

### Update checklist
